### PR TITLE
added report generation

### PR DESCRIPTION
**Pull request by dzhovi**
https://github.com/yegor256/cam/issues/401

### CODE REVIEW

**Comment by dzhovi** [comment]
@yegor256 could you check it? Make job is red again because latex mirror unavailable.

File: steps/report.sh [hunk 1]
```diff
@@ -20,6 +20,10 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
+
+# shellcheck disable=SC2128
```
[the remark this answers is not in the document]
Comment (dzhovi) [reply steps/report.sh]: @yegor256 fixed

File: steps/report.sh [hunk 2]
```diff
@@ -57,6 +58,67 @@ fi
 
 sort -o "${list}" "${list}"
 
+# Create the aggregation table LaTeX file
+aggregation_table=${TARGET}/temp/aggregation_table.tex
+rm -f "${aggregation_table}"
+touch "${aggregation_table}"
```
Comment (yegor256) [inline steps/report.sh:64]: @dzhovi I believe, these two command may be replaced with one: `echo > "${aggregation_table}"`

File: steps/report.sh [hunk 3]
```diff
@@ -57,6 +58,67 @@ fi
...
+  printf "\\\\begin{longtable}{|l|c|c|c|}\n"
+  printf "\\hline\n"
+  printf "Metric & 90th Percentile & Mean & Median \\\\\\\\\\\\\ \n"
+  printf "\\hline\n"
+} >> "${aggregation_table}"
+
+files=("${TARGET}/data/aggregation/*.csv")
+
+# shellcheck disable=SC2128
+# I disabled it, because unwrapping array and take first elem is exactly what i need in this script
```
Comment (yegor256) [inline steps/report.sh:83]: @dzhovi can't you just do it the way shellcheck recommends?

File: steps/aggregate.sh [hunk 4]
```diff
@@ -98,4 +76,24 @@ done < "${repos}"
...
+        output_folder="${TARGET}/data/aggregation"
+        for sh_script in "${LOCAL}/steps/aggregation-functions/"*.sh; do
+            if [[ -f "${sh_script}" ]]; then
+                printf "%s %s %s %s\n" "${sh_script@Q}" "${metric_file}" "${output_folder@Q}" "${metric@Q}" >> "${f_jobs}"
+            fi
+        done
+    fi
+done
+"${LOCAL}/help/parallel.sh" "${f_jobs}"
+wait
```
Comment (yegor256) [inline steps/aggregate.sh:97]: @dzhovi why did you move this block to the bottom of the file? I can't understand the reason for this change

**Review by yegor256 — CHANGES_REQUESTED** [review]
@dzhovi looks very good! see a few comments above

File: steps/aggregate.sh (hunk 4 above)
Comment (dzhovi) [reply steps/aggregate.sh:97]: @yegor256 Aggregation requires csv files for each metric, like LCOM5.csv, which store values for each class. I did not face that problem when i was testing it previously, as these files were generated from previous runs. But doing clear run will fail with a error, so i put this block in the end to have a guarantees over metric files existence

File: steps/report.sh (hunk 2 above)
Comment (dzhovi) [reply steps/report.sh]: @yegor256 fixed

File: steps/report.sh (hunk 3 above)
Comment (dzhovi) [reply steps/report.sh]: @yegor256 fixed

**Comment by dzhovi** [comment]
@yegor256 could you check make job? there are some issues with java 22

**Pull request merged**